tensorflow/stream_executor/platform/default/dso_loader.cc

```diff
@@ -41,6 +46,42 @@ string GetDirectMLPath() {
   return (path != nullptr ? path : "");
 }
 
+#if _WIN32
+string GetModuleDirectory() {
+  HMODULE tensorflowHmodule = nullptr;
+  BOOL getHandleResult = GetModuleHandleExW(
+      GET_MODULE_HANDLE_EX_FLAG_UNCHANGED_REFCOUNT |
+          GET_MODULE_HANDLE_EX_FLAG_FROM_ADDRESS,
+      reinterpret_cast<LPCWSTR>(&GetModuleDirectory), &tensorflowHmodule);
+  CHECK_EQ(getHandleResult, TRUE);
+
+  // Safe const_cast because of explicit bounds checking and contiguous memory
+  // in C++11 and later.
+  std::wstring wpath(MAX_PATH, '\0');
+  DWORD filePathSize =
+      GetModuleFileNameW(tensorflowHmodule, const_cast<wchar_t*>(wpath.data()),
+                         static_cast<DWORD>(wpath.size()));
+
+  // Stop searching if the path is 2^16 characters long to avoid allocating an
+  // absurd amount of memory. Where DID you install python?
+  while ((GetLastError() == ERROR_INSUFFICIENT_BUFFER) &&
+         (wpath.size() < 65536)) {
+    wpath.resize(wpath.size() * 2);
+    filePathSize = GetModuleFileNameW(tensorflowHmodule,
+                                      const_cast<wchar_t*>(wpath.data()),
+                                      static_cast<DWORD>(wpath.size()));
+  }
+  CHECK_NE(filePathSize, 0);
+
+  // Strip TF library filename from the path.
+  CHECK_EQ(
+      PathCchRemoveFileSpec(const_cast<wchar_t*>(wpath.data()), wpath.size()),
```

Comment:
>PathCchRemoveFileSpec [](start = 6, length = 21)

You should explicitly truncate the string afterwards with `wpath.resize(wcslen(wpath.c_str()))`